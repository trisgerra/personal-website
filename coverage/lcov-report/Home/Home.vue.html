<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for Home/Home.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">Home</a> Home.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>7/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>6/6</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">10x</span>
<span class="cline-any cline-yes">4x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;div class="home"&gt;
    &lt;head-hero&gt;&lt;/head-hero&gt;
    &lt;div class="card-container"&gt;
      &lt;div class="columns"&gt;
        &lt;div v-for="card in cards" v-bind:key="card.cardName" class="column home-card"&gt;
          &lt;home-card :card="card"&gt;&lt;/home-card&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
import { Component, Vue } from 'vue-property-decorator';
import HeadHero from './HeadHero/HeadHero.vue';
import HomeCard from './HomeCard/HomeCard.vue';
import { homeCards, ICard } from './homeCardItems';
&nbsp;
@Component({
  components: {
    HeadHero,
    HomeCard,
  },
})
export default class Home extends Vue {
  cards: ICard[] = homeCards;
}
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" scoped&gt;
.columns {
    margin: 0px;
    width: 100%;
}
.column {
    padding:0px;
}
.card-container {
  border-top: 3px solid #FF0000;
  border-right: 3px solid #FF0000;
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 13 2019 00:06:37 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
