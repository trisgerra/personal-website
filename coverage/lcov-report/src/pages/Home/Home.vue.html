<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for src/pages/Home/Home.vue</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../../../prettify.css" />
    <link rel="stylesheet" href="../../../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../../../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../../../index.html">All files</a> / <a href="index.html">src/pages/Home</a> Home.vue
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>9/9</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>0/0</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>1/1</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">100% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>8/8</span>
      </div>
    </div>
  </div>
  <div class='status-line high'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43
44
45
46
47
48
49
50
51
52
53
54
55
56
57
58
59
60
61
62</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">11x</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">6x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">&lt;template&gt;
  &lt;div class="home"&gt;
    &lt;head-hero&gt;&lt;/head-hero&gt;
    &lt;div class="card-container"&gt;
      &lt;div class="columns"&gt;
        &lt;div v-for="card in cards" v-bind:key="card.cardName" class="column home-card"&gt;
          &lt;home-card :card="card"&gt;&lt;/home-card&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
    &lt;div class="square-container"&gt;
      &lt;div v-bind:key="index" v-for="(row, index) in squares" class="square-row"&gt;
        &lt;div class="columns"&gt;
          &lt;div v-for="item in row" v-bind:key="item.title" class="column square-card"&gt;
            &lt;square-card :item="item"&gt;&lt;/square-card&gt;
          &lt;/div&gt;
        &lt;/div&gt;
      &lt;/div&gt;
    &lt;/div&gt;
  &lt;/div&gt;
&lt;/template&gt;
&nbsp;
&lt;script lang="ts"&gt;
import { Component, Vue } from 'vue-property-decorator';
import HeadHero from './HeadHero/HeadHero.vue';
import HomeCard from './HomeCard/HomeCard.vue';
import SquareCard from './SquareCard/SquareCard.vue';
import {
  homeCards, squareCards, ICard, ISquareItem,
} from './homeCardItems';
&nbsp;
@Component({
  components: {
    HeadHero,
    HomeCard,
    SquareCard,
  },
})
export default class Home extends Vue {
  cards: ICard[] = homeCards;
&nbsp;
  squares: ISquareItem[][] = squareCards;
}
&lt;/script&gt;
&nbsp;
&lt;style lang="scss" scoped&gt;
.columns {
  margin: 0px;
  width: 100%;
}
.column {
  padding: 0px;
}
.card-container {
  border-top: 3px solid #ff0000;
  border-right: 3px solid #ff0000;
}
.square-container {
    border-right: 3px solid #ff0000;
}
&lt;/style&gt;
&nbsp;</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Sat Jul 13 2019 23:49:12 GMT+0200 (Central European Summer Time)
</div>
</div>
<script src="../../../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../../../sorter.js"></script>
</body>
</html>
