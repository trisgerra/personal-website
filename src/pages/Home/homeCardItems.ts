interface IItem {
    text: string;
    color: string;
}

export interface ICard {
    name: string;
    items: IItem[],
    isDark: boolean
}

export interface ISquareItem {
  title: string;
  color: string;
  url: string;
}

const experienceList : IItem[] = [
  {
    text: '<b>2012</b> - iOS Internship',
    color: '#FF0000',
  },
  {
    text: '<b>2013</b> - iOS Developer',
    color: '#FF0000',
  },
  {
    text: '<b>2014</b> - Mobile Developer',
    color: '#FF0000',
  },
  {
    text: '<b>2015</b> - Mobile Developer & Analyst',
    color: '#FF0000',
  },
  {
    text: '<b>2016</b> - Mobile Developer, Analyst Designer',
    color: '#FF0000',
  },
  {
    text: '<b>2017</b> - Mobile Lead',
    color: '#FF0000',
  },
  {
    text: '<b>2018</b> - Frontend Developer',
    color: '#00D1FF',
  },
  {
    text: '<b>Now</b> - Senior Frontend Developer',
    color: '#00D1FF',
  },
];

const skillsList : IItem[] = [
  {
    text: '<b>iOS</b> - [Obj-C, Swift, React Native]',
    color: '#FF0000',
  },
  {
    text: '<b>Android</b> - [Java, Swift, React Native]',
    color: '#FF0000',
  },
  {
    text: '<b>Web</b> - [Vue, React]',
    color: '#00D1FF',
  },
  {
    text: '<b>Unit Testing</b> - [Jest]',
    color: '#00D1FF',
  },
  {
    text: '<b>E2E Testing</b> - [Cypress]',
    color: '#00D1FF',
  },
  {
    text: '<b>Design</b> - [Invision Studio, Sketch, Photoshop]',
    color: '#151616',
  },
  {
    text: '<b>CI/CD</b> - [Travis CI, Circle CI, CodeClimate, Jenkins, Heroku]',
    color: '#151616',
  },
  {
    text: '<b>Devops</b> - [Docker, Kubernetes]',
    color: '#151616',
  },
];

export const homeCards : ICard[] = [
  {
    name: 'WORK EXPERIENCE',
    items: experienceList,
    isDark: true,
  },
  {
    name: 'SKILLS',
    items: skillsList,
    isDark: false,
  },
];

export const squareCards : ISquareItem[][] = [
  [{
    title: 'GIT',
    color: '#151616',
    url: 'https://github.com/trisgerra',
  },
  {
    title: 'LINKEDIN',
    color: '#2D2F2F',
    url: 'https://www.linkedin.com/in/a-digiacomo/',
  },
  {
    title: 'PROTEIT',
    color: '#494A4A',
    url: 'https://apps.apple.com/it/app/proteit/id1149272279?l=en',
  },
  {
    title: 'VAPOR',
    color: '#6C6C6C',
    url: 'https://play.google.com/store/apps/details?id=it.asfitness.android.app.vapor.vapor&hl=it',
  }],
  [{
    title: 'BLOG',
    color: '#6C6C6C',
    url: '',
  },
  {
    title: 'FITNESS',
    color: '#494A4A',
    url: '',
  },
  {
    title: 'CONTACT',
    color: '#2D2F2F',
    url: 'mailto:andrea.digiacomo@live.it',
  },
  {
    title: ':)',
    color: '#151616',
    url: '',
  }],
];
