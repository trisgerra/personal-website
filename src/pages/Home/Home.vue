<template>
  <div class="home">
    <head-hero></head-hero>
    <div class="card-container">
      <div class="columns">
        <div v-for="card in cards" v-bind:key="card.cardName" class="column home-card">
          <home-card :card="card"></home-card>
        </div>
      </div>
    </div>
  </div>
</template>

<script lang="ts">
import { Component, Vue } from 'vue-property-decorator';
import HeadHero from './HeadHero/HeadHero.vue';
import HomeCard from './HomeCard/HomeCard.vue';
import { homeCards, ICard } from './homeCardItems';

@Component({
  components: {
    HeadHero,
    HomeCard,
  },
})
export default class Home extends Vue {
  cards: ICard[] = homeCards;
}
</script>

<style lang="scss" scoped>
.columns {
    margin: 0px;
    width: 100%;
}
.column {
    padding:0px;
}
.card-container {
  border-top: 3px solid #FF0000;
  border-right: 3px solid #FF0000;
}
</style>
