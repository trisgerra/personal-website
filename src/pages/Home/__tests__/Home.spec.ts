import { Wrapper } from '@vue/test-utils';
import Home from '../Home.vue';
import createWrapper from '@/utils/CreateWrapper';

describe('Home', () => {
  it('should render as expected', () => {
    const wrapper: Wrapper<Home> = createWrapper(Home);
    expect(wrapper.element).toMatchSnapshot();
  });
});
