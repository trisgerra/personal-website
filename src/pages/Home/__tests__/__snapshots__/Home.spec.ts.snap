// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render as expected 1`] = `
<div
  class="home"
>
  <div
    class="container"
  >
    <div
      class="absolute columns"
    >
      <div>
        <img
          alt="profile image"
          class="profile"
          src="../../../assets/profile.jpg"
        />
      </div>
       
      <div
        class="column text-container"
      >
        <h2
          class="title"
        >
          Welcome to my &lt;space/&gt;
        </h2>
         
        <h3
          class="subtitle"
        >
          WHO AM I?
        </h3>
         
        <p
          class="description"
        >
          I'm a enthusiast Software developer but first of all I consider myself as ethernal curious,&lt;/br&gt; that loves to learn, new things everyday.
        </p>
      </div>
    </div>
     
    <nav
      class="columns"
    >
      <div
        class="column first"
      />
       
      <div
        class="column"
      />
       
      <div
        class="column"
      />
       
      <div
        class="column"
      />
       
      <div
        class="column"
      />
       
      <div
        class="column"
      />
    </nav>
  </div>
   
  <div
    class="card-container"
  >
    <div
      class="columns"
    >
      <div
        class="column home-card"
      >
        <div
          class="card column dark"
        >
          <div
            class="card-content"
          >
            <h3
              class="card-title"
            >
              WORK EXPERIENCE
            </h3>
             
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              2012 - iOS Internship
            </p>
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              2013 - iOS Developer
            </p>
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              2014 - Mobile Developer
            </p>
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              2015 - Mobile Developer & Analyst
            </p>
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              2016 - Mobile Developer, Analyst Designer
            </p>
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              2017 - Mobile Lead
            </p>
            <p
              class="item"
              style="border-left: 3px solid #00D1FF;"
            >
              2018 - Frontend Developer
            </p>
            <p
              class="item"
              style="border-left: 3px solid #00D1FF;"
            >
              Now - Senior Frontend Developer
            </p>
          </div>
        </div>
      </div>
      <div
        class="column home-card"
      >
        <div
          class="card column"
        >
          <div
            class="card-content"
          >
            <h3
              class="card-title"
            >
              SKILLS
            </h3>
             
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              iOS - [Obj-C, Swift, React Native]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #FF0000;"
            >
              Android - [Java, Swift, React Native]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #00D1FF;"
            >
              Web - [Vue, React]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #00D1FF;"
            >
              Unit Testing - [Jest]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #00D1FF;"
            >
              E2E Testing - [Cypress]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #151616;"
            >
              Design - [Invision Studio, Sketch, Photoshop]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #151616;"
            >
              CI/CD - [Travis CI, Circle CI, CodeClimate, Jenkins, Heroku]
            </p>
            <p
              class="item"
              style="border-left: 3px solid #151616;"
            >
              Devops - [Docker, Kubernetes]
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
</div>
`;
